client/relay.go

```diff
@@ -12,35 +12,38 @@ import (
 
 var _ sdk.RelayClient = (*relayClient)(nil)
 
-
+// relayClient is a RelayClient implementation that uses the HTTP client
+// of the Go standard library.
 type relayClient struct {
 	httpClient *http.Client
 }
 
+// NewRelayClient creates a new relay client with the default HTTP client, which
+// is used to send the relay requests.
 func NewRelayClient() (sdk.RelayClient, error) {
 	return &relayClient{
 		httpClient: http.DefaultClient,
 	}, nil
 }
 
+// SendRequest sends the relay request to the supplier at the given URL.
+// It accepts the relay request bytes to avoid relying on protocol specific
+// request objects such as http.Request.
 func (r *relayClient) SendRequest(
 	ctx context.Context,
 	supplierUrlStr string,
-	relayRequestBz []byte,
-	method string,
-	requestHeaders map[string][]string,
+	requestBz []byte,
 ) (relayResponseBz []byte, err error) {
 	supplierUrl, err := url.Parse(supplierUrlStr)
 	if err != nil {
 		return nil, err
 	}
 
-	relayRequestReadCloser := io.NopCloser(bytes.NewReader(relayRequestBz))
+	relayRequestReadCloser := io.NopCloser(bytes.NewReader(requestBz))
 	defer relayRequestReadCloser.Close()
 
 	relayHTTPRequest := &http.Request{
-		Method: method,
```

Comment:
This is better but I still don't fully understand what you envision as the end sate for this.

If we're really going to be "general purpose", then aren't we going to be sending requests directly over TCP and that way `HTTP` is just ONE of the options? 

E.g. see this: https://chatgpt.com/share/61ec49d1-0226-4348-ab0e-98425a1319b6

Reply:
See https://github.com/pokt-network/shannon-sdk/pull/1#discussion_r1621389092